plugins/package-managers/swiftpm/src/main/kotlin/SwiftPm.kt

```diff
@@ -76,7 +76,10 @@ class SwiftPm(
         PackageManagerResult(projectResults, graphBuilder.build(), graphBuilder.packages())
 
     override fun mapDefinitionFiles(definitionFiles: List<File>): List<File> {
-        return definitionFiles.filterNot { file -> file.path.contains(".build/checkouts") }
+        return definitionFiles.filterNot { file ->
+            file.path.contains(".build/checkouts") ||
+                (file.name == PACKAGE_RESOLVED_NAME && file.resolveSibling(PACKAGE_SWIFT_NAME).isFile)
```

Comment:
Shouldn't it be the opposite? Like if both `PACKAGE_SWIFT_NAME` and `PACKAGE_RESOLVED_NAME` (i.e. the lockfile for `PACKAGE_SWIFT_NAME`) and present, we should prefer / only use `PACKAGE_RESOLVED_NAME` as it provides the stable versions?

I mean, if you'd run `swiftpm` in such a directory, if would also install the dependencies from `PACKAGE_RESOLVED_NAME` even if there are later version for the dependencies in `PACKAGE_SWIFT_NAME`. At least that's my understanding.

Reply:
> Shouldn't it be the opposite? Like if both `PACKAGE_SWIFT_NAME` and `PACKAGE_RESOLVED_NAME` (i.e. the lockfile for `PACKAGE_SWIFT_NAME`) and present, we should prefer / only use `PACKAGE_RESOLVED_NAME` as it provides the stable versions?

I think it's better in this case to run `switch packages show-dependencies..` because that command does adhere to the lockfile, but it gives more information: it has the edges of the dependency tree.

> I mean, if you'd run swiftpm in such a directory, if would also install the dependencies from PACKAGE_RESOLVED_NAME even if there are later version for the dependencies in PACKAGE_SWIFT_NAME. At least that's my understanding.

I think the assumption (similar to other package managers) should be that the lockfile `Package.resolved` corresponds to the sibling definition file `Package.swift` and just pins / locks down the dependency versions. 

E.g. analog in NPM world, if there is `package-lock.json` commited, then resolution of deps for `package.son` does adhere to the versions present in the lockfile.

Reply:
So, no change needed for addressing this comment, right?

Reply:
Right.